### PR TITLE
Definition based DML permissions

### PR DESCRIPTION
**Pull request by Page-**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "@types/deep-freeze": "^0.1.1",
     "@types/express": "4.16.0",
     "@types/express-session": "^1.15.12",
-    "@types/lodash": "^4.14.122",
+    "@types/lodash": "4.14.122",
```
Comment (dfunckt) [inline package.json]: Is that deliberate? Why did you need to pin the version?

File: src/sbvr-api/permissions.ts [hunk 2]
```diff
@@ -505,53 +509,132 @@ const createBypassDefinition = (definition: Definition) =>
 		}
 	});
 
+const getAlias = (name: string) => {
+	// TODO-MAJOR: Change $bypass to $permissionbypass or similar
+	if (name.endsWith('$bypass')) {
+		return 'bypass';
+	} else {
```
Comment (dfunckt) [inline src/sbvr-api/permissions.ts]: You don't need the `else` block.

File: src/sbvr-api/permissions.ts [hunk 3]
```diff
@@ -1092,107 +1175,13 @@ const rewriteODataOptions = (
...
-	}
-};
-
 export const addPermissions = Promise.method(
 	(
 		req: PermissionReq,
 		request: ODataRequest & { permissionType?: PermissionCheck },
 	): Promise<void> => {
 		const { vocabulary, resourceName, odataQuery, odataBinds } = request;
+		// const { vocabulary, resourceName, odataBinds } = request;
```
Comment (dfunckt) [inline src/sbvr-api/permissions.ts]: Stray comment.

File: package.json (hunk 1 above)
Comment (Page-) [reply package.json]: It's necessary because it changed `isObject` to be a true type guard so that
```ts
export const resolveOdataBind = (odataBinds: ODataBinds, value: any) => {
	if (_.isObject(value) && value.bind != null) {
		[, value] = odataBinds[value.bind];
	}
	return value;
};
```
now fails because `Property 'bind' does not exist on type 'object'.`
